### DISCUSSION TITLE
DB connection error cause airflow scheduler stop to function

### DISCUSSION THREAD

**Discussion by mingmwang** [apache/airflow#34200] [Q&A]
We are running on airflow 2.6.0 version, the metastore is mySQL 8.x.
And something we see below errors in the airflow-scheduler log and the airflow `scheduler` stop to function.

```python
Traceback (most recent call last):
  File "/opt/nucolumnaradmin/.venv/lib/python3.11/site-packages/sqlalchemy/pool/base.py", line 763, in _finalize_fairy
    fairy._reset(pool, transaction_was_reset)
  File "/opt/nucolumnaradmin/.venv/lib/python3.11/site-packages/sqlalchemy/pool/base.py", line 1038, in _reset
    pool._dialect.do_rollback(self)
  File "/opt/nucolumnaradmin/.venv/lib/python3.11/site-packages/sqlalchemy/engine/default.py", line 683, in do_rollback
    dbapi_connection.rollback()
MySQLdb.OperationalError: (2013, 'Lost connection to server during query') 
```

After restarting the airflow `scheduler` process,  everything comes back. And actually the MySQL instance is running well.

I would like to confirm that is this the expected behavior ?
When the airflow scheduler encounters DB connection errors occasionally, can it recover itself without restarting the process ?

**Comment by potiuk** [comment]
> I would like to confirm that is this the expected behavior ?

Yes

> When the airflow scheduler encounters DB connection errors occasionally, can it recover itself without restarting the process ?

It's the deployment job to restart failed components. That's deliberate decision not to try to recover in case of DB error. because it might leave unused memory and some other side effects. Failing and letting the deployment to take care about it is the cleanest (and pretty standard) approach.

Comment: I think you should try to understand and fix the reason why the connection was dropped by your MySQL server.

I think you should also check your server logs to see what's the root cause. There are several reasons why your MySQL might drop the connection - lack of resources, but also there are other reasons (for example Airflow might generate query that is too long for MySQL to handle and in this case you might want to fine-tune your scheduler (for example decrease this number https://airflow.apache.org/docs/apache-airflow/stable/configurations-ref.html#max-tis-per-query) . But you will only find out by inspecting logs of your MySQL and corellate it with the "lost connection" event to see what was the root cause.
